### PR TITLE
add devnet to github actions and add tests

### PR DESCRIPTION
**Pull request by gregoryguillou**
This PR contributes to #54:

- [x] adds a sidecar the the github action that embeds devnet on 5050
- [x] run a testing script to check the sidecar is up before running tests
- [x] make sure we run some tests on devnet at least with chainid and protocol.

This PR has enabled us to discover issue #66. We now have a devnet on the CI

### CODE REVIEW

File: rpc/client_test.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func TestChainID(t *testing.T) {
 		ChainID string
 	}
 	testSet := map[string][]testSetType{
-		"devnet":  {{ChainID: "DEVNET"}},
+		"devnet":  {{ChainID: "SN_GOERLI"}},
```
Comment (gregoryguillou) [inline rpc/client_test.go:94]: I would think it is a bug on starknet-devnet. I  have open a ticket https://github.com/Shard-Labs/starknet-devnet/issues/146

File: rpc/client.go [hunk 2]
```diff
@@ -58,27 +58,26 @@ func (sc *Client) Close() {
 // ChainID retrieves the current chain ID for transaction replay protection.
 func (sc *Client) ChainID(ctx context.Context) (string, error) {
 	var result string
-	err := sc.c.CallContext(ctx, &result, "starknet_chainId")
+	err := sc.c.CallContext(ctx, &result, "starknet_chainId", []interface{}{}...)
```
Comment (gregoryguillou) [inline rpc/client.go:63]: That is the solution I have found to address #66 . TL;DR is that we would have to talk to the people at Shardlabs to see if that is a bug and can be corrected. At least I did not face the issue with Pathfinder.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -8,6 +8,13 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+
+    services:
+      devnet:
+        image: shardlabs/starknet-devnet:0.2.4
```
Comment (drspacemn) [inline .github/workflows/go.yml]: What's your opinion on pulling latest here instead of v0.2.4?

Comment (gregoryguillou) [reply .github/workflows/go.yml]: I have fixed

**Review by drspacemn — APPROVED** [review]

**Pull request merged**